backend/module/eCampCore/src/EntityService/CampService.php

```diff
@@ -49,7 +48,7 @@ public function __construct(
      * @return ApiProblem|array
      */
     public function fetchByOwner(AbstractCampOwner $owner) {
-        $q = parent::findCollectionQueryBuilder(Camp::class, 'row');
+        $q = parent::findCollectionQueryBuilder(Camp::class, 'row', null);
```

Comment:
Wenn das vormals funktioniert hat, wieso müssen wir diesen Parameter jetzt hinzufügen? Ich fände es schöner wenn die base implementation den Parameter nicht akzeptiert, und die überschriebene Version auf dem UserService den Parameter als optional (`protected function findCollectionQueryBuilder($className, $alias, $params = [])`) akzeptiert. Das ist in OOP erlaubt.

Reply:
Ich glaube dein Vorschlag funktioniert nicht.
Die Methode "findCollectionQueryBuilder" wird in der Basis-Klasse ebenfalls mit allen 3 Argumenten aufgerufen (siehe fetchAllQueryBuilder). Dort wird $params mitgegeben - was einem überschreiber erlaubt diese zu verwenen (z.B. im UserService).

Ich sehe das so:
Beim Erweitern von "findCollectionQueryBuilder" auf 3 Argumente (hinzufügen von  $params) wurde der CampService nicht angepasst.
PHP scheint hier tolerant zu sein - eher zu tolerant.

Reply:
Du hast recht. In dem Fall könnte man höchstens ein Default-Wert von `$params = []` auf dem findCollectionQueryBuilder vom AbstractEntityService definieren.